server/odc-service/src/main/java/com/oceanbase/odc/service/db/DBPLModifyHelper.java

```diff
@@ -99,7 +99,7 @@ private EditPLResp executeWrappedEditPL(String sessionId, EditPLReq editPLReq,
             DBObjectType plType, String tempPlName) throws Exception {
         String plName = editPLReq.getObjectName();
         String editPLSql = editPLReq.getSql();
-        String tempPLSql = editPLSql.replaceFirst(plName, tempPlName);
+        String tempPLSql = editPLSql.replaceFirst(plName.replaceAll("([\\\\+*?\\[\\](){}|.^$])", "\\\\$1"), tempPlName);
```

Comment:
It is recommended to write the regular expression as a separate attribute

Reply:
how about this？
![image](https://github.com/user-attachments/assets/1a8449aa-15d1-4f97-b356-d9ff2b55a50e)
![image](https://github.com/user-attachments/assets/e9cf9c55-f456-4dad-b54b-658ebdbd4398)

Reply:
1. `REGULAR_CHARS` can be `String[]`?
2. It is better not to return `null`, you can return `Optional`

Reply:
private static final char[] REGULAR_CHARS = { '.', '*', '+', '?', '^', '$', '[', ']', '(', ')', '{', '}', '|', '\\' };

Reply:
private static final char[] REGULAR_CHARS = { '.', '*', '+', '?', '^', '$', '[', ']', '(', ')', '{', '}', '|', '\\' };

Reply:
2. return ""
![image](https://github.com/user-attachments/assets/de95b602-4349-4e98-8bad-3929ebff1581)